ambry-router/src/main/java/com.github.ambry.router/AdaptiveOperationTracker.java

```diff
@@ -100,14 +110,97 @@ public void onResponse(ReplicaId replicaId, TrackedRequestFinalState trackedRequ
    * Gets the {@link Histogram} that tracks request latencies to the class of replicas (intra or inter DC) that
    * {@code replicaId} belongs to.
    * @param replicaId the {@link ReplicaId} whose request latency is going to be tracked.
-   * @return the {@link Histogram} that tracks requests to the class of replicas (intra or inter DC) that
-   * {@code replicaId} belongs to.
+   * @param routerConfig the {@link RouterConfig} that specifies which scope the histogram is associated with.
+   * @return the {@link Histogram} associated with this replica.
+   */
+  Histogram getLatencyHistogram(ReplicaId replicaId, RouterConfig routerConfig) {
+    boolean isLocalReplica = replicaId.getDataNodeId().getDatacenterName().equals(datacenterName);
+    Histogram histogramToReturn;
+    // TODO add support for replica-level/disk-level/node-level histogram based on the config
+    switch (routerConfig.routerOperationTrackerMetricScope) {
+      case ColoWide:
+        histogramToReturn = isLocalReplica ? localColoHistogram : crossColoHistogram;
+        break;
+      case PartitionLevel:
+        PartitionId partitionId = replicaId.getPartitionId();
+        histogramToReturn = isLocalReplica ? localColoPartitionToHistogram.get(partitionId)
+            : crossColoPartitionToHistogram.get(partitionId);
+        break;
+      default:
+        throw new IllegalArgumentException("Unsupported operation tracker metric scope.");
+    }
+    return histogramToReturn;
+  }
+
+  /**
+   * Get certain colo-wide latency histogram based on given arguments.
+   * @param routerMetrics the {@link NonBlockingRouterMetrics} that used in this tracker.
+   * @param routerOperation the {@link RouterOperation} that uses this tracker.
+   * @param isLocalColo {@code true} if local latency histogram should be returned. {@code false} otherwise.
+   * @return colo-wide latency histogram.
+   */
+  private Histogram getColoWideTracker(NonBlockingRouterMetrics routerMetrics, RouterOperation routerOperation,
```

Comment:
why is routerMetrics passed into these two methods?

Reply:
Add `routerMetrics` as class member and remove it from methods.